erizo/src/erizo/media/SyntheticInput.cpp

```diff
@@ -0,0 +1,223 @@
+  time_point now = clock_->now();
+  if (now >= next_audio_frame_time_) {
+    sendAudioFrame(audio_frame_size_);
+    next_audio_frame_time_ += audio_period_;
+  }
+  if (now >= next_video_frame_time_) {
+    bool is_keyframe = false;
+    size_t frame_size = getRandomValue(video_avg_frame_size_, video_dev_frame_size_);
+    if (now - last_video_keyframe_time_ > kDefaultVideoKeyframePeriod || keyframe_requested_) {
+      is_keyframe = true;
+      frame_size = getRandomValue(video_avg_keyframe_size_, video_dev_keyframe_size_);
+    }
+    while (frame_size > kMaxPacketSize) {
+      sendVideoframe(is_keyframe, false, kMaxPacketSize);
+      is_keyframe = false;
+      frame_size = frame_size - kMaxPacketSize;
+    }
+    sendVideoframe(is_keyframe, true, frame_size);
+
+    next_video_frame_time_ += video_period_;
+  }
+  now = clock_->now();
+  if ((next_video_frame_time_ <= now || next_audio_frame_time_ <= now) && consecutive_ticks_ < kMaxConsecutiveTicks) {
+    consecutive_ticks_++;
+    tick();
+  } else {
+    consecutive_ticks_ = 0;
+  }
+}
+
+int SyntheticInput::sendPLI() {
+  keyframe_requested_ = true;
+  return 0;
+}
+
+void SyntheticInput::sendVideoframe(bool is_keyframe, bool is_marker, uint32_t size) {
+  erizo::RtpHeader *header = new erizo::RtpHeader();
+  header->setSeqNumber(video_seq_number_++);
+  header->setTimestamp(ClockUtils::timePointToMs(clock_->now()) * kVideoSampleRate / 1000);
+  header->setSSRC(video_ssrc_);
+  header->setMarker(is_marker);
+  header->setPayloadType(video_pt_);
+  char packet_buffer[kMaxPacketSize];
+  memset(packet_buffer, 0, size);
+  char* data_pointer;
+  char* parsing_pointer;
+  memcpy(packet_buffer, reinterpret_cast<char*>(header), header->getHeaderLength());
+  data_pointer = packet_buffer + header->getHeaderLength();
+  parsing_pointer = data_pointer;
+  *parsing_pointer = 0x10;
+  parsing_pointer++;
+  *parsing_pointer = is_keyframe ? 0x00 : 0x01;
+
+  if (is_keyframe) {
+    last_video_keyframe_time_ = clock_->now();
+    keyframe_requested_ = false;
+  }
+  if (videoSink_) {
+    videoSink_->deliverVideoData(packet_buffer, size);
+  }
+  delete header;
+}
+
+void SyntheticInput::sendAudioFrame(uint32_t size) {
+  erizo::RtpHeader *header = new erizo::RtpHeader();
+  header->setSeqNumber(audio_seq_number_++);
+  header->setTimestamp(ClockUtils::timePointToMs(clock_->now()) * (kAudioSampleRate / 1000));
+  header->setSSRC(audio_ssrc_);
+  header->setMarker(true);
+  header->setPayloadType(audio_pt_);
+  char packet_buffer[kMaxPacketSize];
+  memset(packet_buffer, 0, size);
+  memcpy(packet_buffer, reinterpret_cast<char*>(header), header->getHeaderLength());
+  if (audioSink_) {
+    audioSink_->deliverAudioData(packet_buffer, size);
+  }
+  delete header;
+}
+
+void SyntheticInput::close() {
+  running_ = false;
+}
+
+void SyntheticInput::calculateSizeAndPeriod(uint32_t video_bitrate, uint32_t audio_bitrate) {
+  ELOG_TRACE("Bitrate received: %d", video_bitrate);
```

Comment:
Actually I'll remove it since it's not needed anymore. Just used it to check that everything worked as expected during my manual tests.